Microsoft.Azure.Cosmos/src/Headers/Headers.cs

```diff
@@ -36,6 +36,15 @@ public virtual string ContinuationToken
             internal set => this.CosmosMessageHeaders.Continuation = value;
         }
 
+        /// <summary>
+        /// Gets or Set the CoorelatedActivityId in the current <see cref="ResponseMessage"/>.
+        /// </summary>
+        internal virtual string CorrelatedActivityId
+        {
+            get => this.CosmosMessageHeaders.INameValueCollection?.Get(HttpConstants.HttpHeaders.CorrelatedActivityId);
```

Comment:
The Get should be available I believe

```suggestion
            get => this.CosmosMessageHeaders.Get(HttpConstants.HttpHeaders.CorrelatedActivityId);
```